### PR TITLE
Add MaxRicciScalarTag to Tags.hpp

### PR DESCRIPTION
**Pull request by MarloMo**
## Proposed changes

The MaxRicciScalarTag and MaxRicciScalarTagCompute calculate the pointwise maximum of the Strahlkorper's intrinsic Ricci scalar curvature.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by geoffrey4444 — DISMISSED** [review]

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [inline src/ApparentHorizons/Tags.hpp:377]: This may be me forgetting how AH stuff works, but why is this templated on Frame but the simple tag is not?

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [inline src/ApparentHorizons/Tags.hpp:383]: This is not absolutely trivial, so it should be tested.

**Review by wthrowe — CHANGES_REQUESTED** [review]
About the commit message:
1. There is no "MaxRicciScalarTag".  You called it MaxRicciScalar.
2. There are 43 files called Tags.hpp in SpECTRE, so that part is not very informative.  I'd just leave it out and let people look at the diff if they want to know the details.

File: src/ApparentHorizons/Tags.hpp
Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp:377]: I'm not sure, I will consult with Dr. Lovelace and I will get back to you.

Comment (geoffrey4444) [reply src/ApparentHorizons/Tags.hpp:377]: Please try removing line 377 and let me know if you get a compiler error

File: src/ApparentHorizons/Tags.hpp
Comment (geoffrey4444) [reply src/ApparentHorizons/Tags.hpp:383]: Maybe an easy way to test this would be to make a function in Test_Tags.cpp that makes a small DataVector and then tests that MaxRicciScalar::function() in fact returns the max?

File: src/ApparentHorizons/Tags.hpp
Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp:377]: When I take the Frame out, I get a complaint from Test_tags.hpp, I'm not sure how to change the code to not take a Frame here. I get. 

`error: expected '>'
      StrahlkorperTags::MaxRicciScalarCompute<Frame::Inertial>>(`

It complains about the <Frame here though it is out of the code. 

This is what I have in Test_Tags.hpp 

`TestHelpers::db::test_compute_tag <
      StrahlkorperTags::MaxRicciScalarCompute("MaxRicciScalar");`

@geoffrey4444

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [reply src/ApparentHorizons/Tags.hpp:383]: I think that would be fine.  You can put the DataVector in a DataBox and fully use the compute item if you want to be really thorough.

File: src/ApparentHorizons/Tags.hpp
Comment (wthrowe) [reply src/ApparentHorizons/Tags.hpp:377]: You've removed one too many '>' characters.  You want
`TestHelpers::db::test_compute_tag<StrahlkorperTags::MaxRicciScalarCompute>("MaxRicciScalar");`.

Comment (MarloMo) [reply src/ApparentHorizons/Tags.hpp:377]: Thank you, @wthrowe , everything compiled fine, I'll move on to the test.

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (wthrowe) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: Just write `3.` here.  In a test, it is best to just write in the answer if it is easy to find.  This eliminates the possibility that there could be a bug in the code that calculates the expected value.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Except for my minor comment, I am happy with this test.

The new commits are only adding code, so I think you can squash everything into one commit without getting @geoffrey4444's approval first.

**Review by wthrowe — DISMISSED** [review]

**Review by geoffrey4444 — DISMISSED** [review]

File: src/ApparentHorizons/Tags.hpp
Comment (nilsvu) [inline src/ApparentHorizons/Tags.hpp]: @kidder Do you require compute tags return by reference so soon that this one should be updated now, or should we merge as-is and change later?

Comment (nilsvu) [reply src/ApparentHorizons/Tags.hpp]: @MarloMo In case @kidder wants you to change this now, all you need to do is this:

```cpp
static void function(const gsl::not_null<double*> max_ricci_scalar, const Scalar<DataVector>& ricci_scalar) noexcept {
  *max_ricci_scalar = max(get(ricci_scalar));
}
```

You might as well just do that now (without waiting for @kidder's response), and squash the change right away.

Comment (kidder) [reply src/ApparentHorizons/Tags.hpp]: yes, please do this

**Review by kidder — CHANGES_REQUESTED** [review]
Asking for a change so this doesn't get merged yet (please make the change @nilsleiffischer asked for)

File: tests/Unit/ApparentHorizons/Test_Tags.cpp
Comment (kidder) [inline tests/Unit/ApparentHorizons/Test_Tags.cpp]: You'll need to change this line to 
```
double max;
StrahlkorperTags::MaxRicciScalarCompute::function(make_not_null(&max), d);
```

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
looks good, you can squash

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**
